processing/src/main/java/io/druid/query/MetricsEmittingExecutorService.java

```diff
@@ -23,14 +23,49 @@
 import com.metamx.emitter.service.ServiceEmitter;
 import com.metamx.emitter.service.ServiceMetricEvent;
 
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadMXBean;
 import java.util.concurrent.Callable;
 
 public class MetricsEmittingExecutorService extends ForwardingListeningExecutorService
 {
+  private static final ThreadMXBean THREAD_MX_BEAN = ManagementFactory.getThreadMXBean();
+  private static final ThreadLocal<Long> THREAD_USER_START = new ThreadLocal<Long>()
+  {
+    @Override
+    public Long initialValue()
+    {
+      return 0L;
+    }
+  };
+  private static final Runnable PRE = new Runnable()
+  {
+    @Override
+    public void run()
+    {
+      if (THREAD_MX_BEAN.isCurrentThreadCpuTimeSupported()) {
```

Comment:
I originally wasn't really sure at what level the bean would be consistent, as in: could you have some threads in a JVM that can support cpu time but not others? or could you have a flag enabled during runtime that could enable/disable CPUTime?

It looks like the Sun/Oracle way of doing things is to have the value assumed to not change per whole JVM, but I'm not sure if there's an official contract somewhere that enforces such a thing.

The proposed change makes sense though, and is pretty darn straight forward, so I'll put it in even though it is optimization without profiling.